llama_index/llms/openai_like.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any
+
+from llama_index.bridge.pydantic import Field
+from llama_index.llms import OpenAI
+
+
+class OpenAILike(OpenAI):
+    context_window: int = Field(
+        default=512, discription="Custom context_window for custom models."
+    )
+    is_chat_model: bool = Field(
+        default=False, description="Indicates that the custom model is a chat_model."
+    )
+    is_function_calling_model: bool = Field(
```

Comment:
I think you need to override `def metadata` to make sure this `Field` gets used, currently it's dead code.

Might be nice to add an assertion for its use in `test_interfaces`